ui/desktop/app/routes/scopes/scope/projects/sessions/index.js

```diff
@@ -97,43 +97,70 @@ export default class ScopesScopeProjectsSessionsIndexRoute extends Route {
       page,
       pageSize,
       force_refresh: true,
+      max_result_set_size: -1,
     };
     if (orgScope.isOrg && scopes.length === 0) {
       queryOptions.filter = orgFilter;
     }
     const sessions = await this.store.query('session', queryOptions);
     const totalItems = sessions.meta?.totalItems;
 
-    // Query all sessions and all targets for defining filtering values if entering route for the first time
+    // Query all sessions and associated targets for defining filtering values if entering route for the first time
     if (!this.allSessions?.length) {
       await this.getAllSessions(orgScope, scopes, orgFilter);
     }
-    if (!this.allTargets?.length) {
-      await this.getAllTargets(orgScope, orgFilter);
-    }
+    const associatedTargets = await this.getAssociatedTargets(
+      orgScope,
+      orgFilter,
+    );
 
     return {
       sessions,
       projects,
       allSessions: this.allSessions,
-      allTargets: this.allTargets,
+      associatedTargets,
       totalItems,
       isCacheDaemonRunning: await this.ipc.invoke('isCacheDaemonRunning'),
     };
   }
 
-  async getAllTargets(orgScope, orgFilter) {
-    const allTargetsQuery = {
-      scope_id: orgScope.id,
-      recursive: true,
-      force_refresh: true,
-    };
-    if (orgScope.isOrg) {
-      allTargetsQuery.filter = orgFilter;
+  async getAssociatedTargets(orgScope) {
+    if (!this.allSessions) {
+      return [];
     }
-    this.allTargets = await this.store.query('target', allTargetsQuery, {
-      pushToStore: false,
-    });
+
+    const uniqueTargetIds = [
+      ...new Set(this.allSessions.map((session) => session.target_id)),
+    ];
+    // Chunk target IDs in case there's a lot of sessions as cache daemon syntax can't take too many
+    const chunkedTargetIds = chunk(uniqueTargetIds, 50);
```

Comment:
If im understanding this correctly, we are separating the target ids into chunks of 50, why not 250 since that is the limit ? Is it because its faster to do more smaller chunks than less bigger chunks ?